src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -51,12 +53,18 @@ public DefaultProcessRunner(ProcessBuilder processBuilder) {
    *
    * @param command The shell command to execute
    */
-  public void run(String[] command) throws ProcessRunnerException {
+  public synchronized void run(String[] command) throws ProcessRunnerException {
```

Comment:
There is a race condition between checking if the process is null and starting one. I should probably scope the synchronization to the affected part of the method only. Doing it on the whole method also has the effect of blocking other threads when they call run and there is a synchronous execution in progress. It's probably not the behavior we want.